include/aws/s3/s3_client.h

```diff
@@ -202,6 +202,22 @@ struct aws_s3_client_config {
     void *shutdown_callback_user_data;
 };
 
+/* Keepalive properties are TCP only.
+ * If interval or timeout are zero, then default values are used.
+ */
+struct aws_s3_tcp_keep_alive_options {
+
+    /* Set keepalive true to periodically transmit messages for detecting a disconnected peer.*/
+    bool keepalive;
```

Comment:
do we need bool here? wouldnt null keep alive indicate keep alive is on?

Reply:
Do you mean null keep alive will indicate that keep alive is off? Since it is just a normal struct, it can't be null. We can turn it into a pointer and remove the keepalive. If the pointer is null, that would mean keepalive is off and vice versa. 

Otherwise, we need a keepalive variable so that users can just set keepalive to true and use default values for the rest. This is basically a partial struct from aws_socket_options which has two extra fields  enum aws_socket_type type and enum aws_socket_domain domain. I didn't want to allow users to override these two fields.